hack/generate-parallel.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+CURRENT_DIR="$(dirname $0)"
+PROJECT_ROOT="${CURRENT_DIR}"/..
+if [ "${PROJECT_ROOT#/}" == "${PROJECT_ROOT}" ]; then
+  PROJECT_ROOT="./$PROJECT_ROOT"
+fi
+
+pushd "$PROJECT_ROOT" > /dev/null
+ROOTS=${ROOTS:-$(git grep --files-with-matches -e '//go:generate' cmd extensions pkg plugin test | \
+	xargs -n 1 dirname | \
+	sed 's,^,github.com/gardener/gardener/,;' | \
```

Comment:
These scripts in the `hack` dir are also used by extensions, so this here should probably be configurable.

Reply:
Yes, in fact I thought we should perhaps preserve the original `generate` script as it is for use by extensions, and use `generate-parallel.sh` only here. For extensions code generation is fast enough from my perspective and there is little to be gained by parallelisation, further more as pointed out above it's by no means guaranteed that whatever you put in `//go:generate` directives can be executed in parallel. I had to replace some `go:generate` by explicit invocations in the `Makefile` to make it work.

Reply:
OK, true, makes sense. There is not much to gain and it's not likely that this will change in the near future.